core/trino-main/src/main/java/io/trino/operator/aggregation/DecimalSumAggregation.java

```diff
@@ -120,16 +122,20 @@ public static void outputDecimal(@AggregationState LongDecimalWithOverflowState
     {
         if (state.isNotNull()) {
             if (state.getOverflow() != 0) {
-                throw new ArithmeticException("Decimal overflow");
+                throw new TrinoException(NUMERIC_VALUE_OUT_OF_RANGE, "Decimal overflow");
             }
 
             long[] decimal = state.getDecimalArray();
             int offset = state.getDecimalArrayOffset();
 
             long rawHigh = decimal[offset];
             long rawLow = decimal[offset + 1];
-
-            Decimals.throwIfOverflows(rawHigh, rawLow);
+            try {
+                Decimals.throwIfOverflows(rawHigh, rawLow);
```

Comment:
Let's inline this method and throw the proper exception directly. Then mark the method as `@Deprecated(forRemoval = true)` since the only caller is in the same class.

Reply:
oh. nice. will do. thanks.

Reply:
@electrum, do you still want me to update this given that it is being called in two places?

Reply:
Yes, please inline the existing usages and mark the method as deprecated. I think that will make the code clearer.